services/ymax-planner/src/gas-estimation.ts

```diff
@@ -0,0 +1,78 @@
+import type { AxelarChain } from '@agoric/portfolio-api/src/constants';
+import { gasLimitEstimates } from './support.ts';
+
+const AGORIC_CHAIN = 'agoric';
+const BLD_TOKEN = 'ubld';
+
+export type GasEstimator = ReturnType<typeof makeGasEstimator>;
+
+export const makeGasEstimator = ({
+  axelarApiAddress,
+  fetchFunc,
+  log,
+}: {
+  axelarApiAddress: string;
+  fetchFunc: typeof fetch;
+  log: typeof console.log;
+}) => {
+  const queryAxelarGasAPI = async (
+    sourceChain: AxelarChain | 'agoric',
+    destinationChain: AxelarChain | 'agoric',
+    gasLimit: bigint,
+    gasToken?: string,
+  ) => {
+    const response = await fetchFunc(axelarApiAddress, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        sourceChain,
+        destinationChain,
+        gasLimit: `${gasLimit}`,
+        sourceTokenSymbol: gasToken,
+        gasMultiplier: '1',
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+
+    const body = await response.json();
+    try {
+      return BigInt(body);
+    } catch (err) {
+      log('⚠️ Axelar gas API non-integer response, trying to round up', body);
+      throw err;
+    }
```

Comment:
This won't work as expected; the raw HTTP response body seems to be a string of digits that `.json()` will [potentially lossily] convert to a number.
```suggestion
    const body = await response.text();
    return BigInt(body.trim());
```